packages/mobx/src/api/makeObservable.ts

```diff
@@ -53,8 +54,13 @@ export function makeAutoObservable<T extends object, AdditionalKeys extends Prop
             die(`makeAutoObservable can only be used on objects not already made observable`)
     }
 
-    const adm: ObservableObjectAdministration = asObservableObject(target, options)[$mobx]
+    // Optimization (avoids visiting protos)
+    // assumes that annotation.make_/.extend_ works the same for plain objects
+    if (Object.getPrototypeOf(target) === objectPrototype) {
```

Comment:
The proto not being set should take the same path. (E.g. objects created with `Object.create({}, null)`)